ls1intum/Artemis
# `Communication`: Disable notifications for hidden conversations

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


Users receive notifications for conversations that are hidden by a user, which can be annoying for a user since they do not seem to be interested in the conversation.

### Description

If a conversation is hidden, users don't receive notifications anymore, except it is an announcement channel.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Instructor
- Conversations of each type (standard channel, course wide channel, group chat, one to one chat)

1. Login as the student
2. Hide each of the conversations
3. Leave the messages tab with the student
4. Login as the instructor as well
5. Write messages as the instructor, the student should not see any notifications
6. Leave the messages tab with the instructor
7. Write messages as the student, the instructor should see notifications
8. Create a new lecture with a visible date in the future
9. Write messages within the lecture channel as the instructor, the student should not see notifications

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [ConversationWebSocketRecipientSummary.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5339/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/ConversationWebSocketRecipientSummary.html) | 85% | ✅ |
| [UserRepository.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5339/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.repository/UserRepository.html) | 91% | ✅ |
| [ConversationMessagingService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5339/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.metis/ConversationMessagingService.html) | 87% | ✅ |
| [PostingService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5339/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.metis/PostingService.html) | 99% | ✅ |
| [ConversationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5339/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.metis.conversation/ConversationService.html) | 96% | ✅ |

### Screenshots


No UI changes
